src/main/java/org/jnode/fs/xfs/XfsDirectory.java

```diff
@@ -121,7 +120,7 @@ protected FSEntryTable readEntries() throws IOException {
 
             case EXTENTS:
                 if (!inode.isDirectory()) {
-                    throw new UnsupportedOperationException("Trying to get directories of a non directory inode");
+                    throw new UnsupportedOperationException("Trying to get directories of a non directory inode (node mode " + inode.getMode() + ")");
```

Comment:
I can't reproduce the issue yet, but I think it is good to get what it might be.